lib/db_connection.ex

```diff
@@ -423,6 +423,7 @@ defmodule DBConnection do
 
     * `{:connected, pid}`
     * `{:disconnected, pid}`
+    * `{:queue_timeout, %DBConnection.ConnectionError{}}`
```

Comment:
Oh, one last question, does the error include all relevant information, such as the options and repository that failed?

Reply:
No, but the options passed to the `checkout` function is something that we can easily add. The repo itself I'm not sure, but in the worst case scenario, multi-repo systems can have one listener for each repo.

Inspection `opts` prints:

```elixir
[
  agent: #PID<0.292.0>,
  parent: #PID<0.291.0>,
  backoff_start: 30000,
  queue_timeout: 10,
  queue_target: 10,
  queue_interval: 10,
  connection_listeners: [#PID<0.291.0>],
  pool: DBConnection.ConnectionPool,
  pool_size: 1
]
```

Reply:
No worries! I could see both ideas making sense, at least we explored both options :D
Should I refactor this to publish the telemetry event then?
Which metadata should I include?

Reply:
I would include the whole exception and the options too.

Reply:
you can send the telemetry one as a separate PR, so we have both as reference!